homeassistant/components/homematicip_cloud/sensor.py

```diff
@@ -183,10 +187,74 @@ async def async_setup_entry(
                     if ch.currentPowerConsumption is not None:
                         entities.append(HmipEsiLedCurrentPowerConsumption(hap, device))
                     entities.append(HmipEsiLedEnergyCounterHighTariff(hap, device))
+        if isinstance(device, AsyncFloorTerminalBlock12):
+            for channel in device.functionalChannels:
+                if isinstance(channel, FloorTerminalBlockMechanicChannel):
+                    if getattr(channel, "valvePosition", None) is not None:
+                        entities.append(
+                            HomematicipFloorTerminalBlockMechanicChannelValve(
+                                hap, device, channel=channel.index
+                            )
+                        )
```

Comment:
I think what helps with simplifying is to make this a single list comprehension

Reply:
Done in https://github.com/home-assistant/core/pull/122759/commits/03b46cd5a6cc0f406f3f09919b5947c34ad10cd1